### PR TITLE
update document for release 2.2.0

### PR DESCRIPTION
**Pull request by marsishandsome**
### What problem does this PR solve? 
update document for release 2.2.0

### CODE REVIEW

File: docs/datasource_api_userguide.md [hunk 1]
```diff
@@ -308,7 +300,9 @@ The following table shows the TiDB-specific options, which can be passed in thro
 
 ## TiDB Version and Configuration for Write
 
-TiDB's version must be 3.0 or later.
+TiDB's version must be 4.0 or later.
+
+**IMPORTANT: currently TiDB-4.0 is not released yet, but you can use code on tidb's master branch.**
```
Comment (zhexuany) [inline docs/datasource_api_userguide.md]: s/can use code on tidb's master branch./can use tidb's master branch. 

Additionally, any requirement of TiKV and pd?

File: docs/datasource_api_userguide.md [hunk 2]
```diff
@@ -324,7 +318,7 @@ delay-clean-table-lock: 60000
 split-table: true
 ```
 
-If your TiDB's version is earlier than 3.0, set `spark.tispark.write.without_lock_table` to `true` to enable write, but ACID is **not** guaranteed.
+If your TiDB's version is earlier than 4.0, set `spark.tispark.write.without_lock_table` to `true` to enable write, but ACID is **not** guaranteed.
```
Comment (zhexuany) [inline docs/datasource_api_userguide.md]: Do we need to fill more info here? @birdstorm

File: docs/datasource_api_userguide.md (hunk 1 above)
Comment (marsishandsome) [reply docs/datasource_api_userguide.md]: fixed.

requirement of TiKV and pd is that: compatible with tidb master branch

File: docs/datasource_api_userguide.md (hunk 2 above)
Comment (birdstorm) [reply docs/datasource_api_userguide.md]: We may add the detailed info when we throw exception in batch write?

Comment (birdstorm) [reply docs/datasource_api_userguide.md]: We may add the detailed info when we throw exception in batch write?

**Pull request merged**